### PR TITLE
Add centeredMessage to GeneralExtensions

### PR DESCRIPTION
**Pull request by kxmpxtxnt**

### CODE REVIEW

**Comment by kxmpxtxnt** [comment]
sorry, forgot something

**Comment by kxmpxtxnt** [comment]
nvm, i forgot nothing, im just dumb

**Comment by kxmpxtxnt** [comment]
someone took my confidence
I look again

**Comment by jakobkmar** [comment]
Don't close the PR, draft PRs are there for this.

**Comment by kxmpxtxnt** [comment]
Okay

**Comment by kxmpxtxnt** [comment]
what do you think of the idea so far?

**Comment by mooziii** [comment]
nice idea but won't work properly because of this setting 

 img

**Comment by kxmpxtxnt** [comment]
yes, unfortunately you can only do this with the default value...
I hope that spigot will eventually release a function that allows to get the chatsize

**Comment by jakobkmar** [comment]
Spigot cannot provide such a function since the client does not send that information to the server.

**Comment by icrayix** [comment]
Spigot can't add functionality for that as such settings aren't sent to the server

**Comment by kxmpxtxnt** [comment]
this is the problem

**Comment by kxmpxtxnt** [comment]
so you can look over it again, now several lines etc should work for the time being

**Comment by kxmpxtxnt** [comment]
I mean, there is already Player#getClientOption(ClientOption.CHAT_VISIBILITY), for example, so maybe something like the size will be included soon.

**Comment by kxmpxtxnt** [comment]
yes unfortunately, but hope dies last xd

**Comment by kxmpxtxnt** [comment]
yes unfortunately, but hope dies last xd

**Comment by kxmpxtxnt** [comment]
would be something for Pacmc that the modloader always sends this value per packet to the server :D

**Comment by kxmpxtxnt** [comment]
> would be something for Pacmc that the "client" always sends this value per packet to the server :D

you don't have to answer, it's wishful thinking

**Comment by kxmpxtxnt** [comment]
Should I then close the pr or is there any hope that it will be implemented

**Comment by jakobkmar** [comment]
I wouldn't merge this until there is a way to get the chat width on the server. As long is that is not possible, this feature is way too unstable.

**Comment by kxmpxtxnt** [comment]
Ok, that's fine, too bad

**Comment by kxmpxtxnt** [comment]
would it be an idea if we create an insecure/instable branch or something where we put such prs in that people who want to use it can take it?

**Comment by 0utplay** [comment]
People who need such features can create them themselves. Maintaining multiple branches is unnecessary work in my opinion

**Comment by kxmpxtxnt** [comment]
understandable

File: src/main/kotlin/net/axay/kspigot/extensions/GeneralExtensions.kt [hunk 1]
```diff
@@ -64,3 +64,160 @@ fun pluginKey(key: String) = NamespacedKey(PluginInstance, key)
...
+ * @return Component
+ * @param content Content which will centered
+ */
+fun centeredMessage(content: String): Component {
+  val center_px = 154
+  var message_px = 0
+  var previousCode = false
+  var isBold = false
+
+  for(c in content.toCharArray()){
```
Comment (Krxwallo) [inline src/main/kotlin/net/axay/kspigot/extensions/GeneralExtensions.kt]: Reformat pls

**Comment by r0yzer** [comment]
genial

**Comment by FlerBuster** [comment]
flerdammt

**Pull request closed without merging**